qmi/instruments/thorlabs/mpc320.py

```diff
@@ -0,0 +1,507 @@
+        # Get response
+        resp = self._apt_protocol.ask(MOD_GET_CHANENABLESTATE)
+        # For the MPC320 the state 0x01 is the ENABLE state and anything else is DISABLE
+        if resp.enable_state == 0x01:
+            return AptChannelState.ENABLE
+        return AptChannelState.DISABLE
+
+    @rpc_method
+    def start_auto_status_update(self) -> None:
+        """
+        Start automatic status updates from device.
+        """
+        _logger.info("[%s] Starting automatic status updates from instrument", self._name)
+        self._check_is_open()
+        # Send message.
+        self._apt_protocol.write_param_command(AptMessageId.HW_START_UPDATEMSGS.value)
+
+    @rpc_method
+    def stop_auto_status_update(self) -> None:
+        """
+        Stop automatic status updates from device.
+        """
+        _logger.info("[%s] Stopping automatic status updates from instrument", self._name)
+        self._check_is_open()
+        # Send message.
+        self._apt_protocol.write_param_command(AptMessageId.HW_STOP_UPDATEMSGS.value)
+
+    @rpc_method
+    def home_channel(self, channel_number: int) -> None:
+        """
+        Start the homing sequence for a given channel.
+
+        Paramters:
+            channel_number: The channel to home.
+        """
+        _logger.info("[%s] Homing channel %d", self._name, channel_number)
+        self._validate_channel(channel_number)
+        self._check_is_open()
+        # Send message.
+        self._apt_protocol.write_param_command(
+            AptMessageId.MOT_MOVE_HOME.value, Thorlabs_MPC320_ChannelMap[channel_number]
+        )
+
+    @rpc_method
+    def is_channel_homed(self, channel_number: int, timeout: float = DEFAULT_RESPONSE_TIMEOUT) -> bool:
+        """
+        Check if a given channel is homed. This command should only be run after the method `home_channel`.
+        Otherwise you will read bytes from other commands using this method.
+
+        Paramters:
+            channel_number: The channel to check.
+            timeout:        The time to wait for a response to the homing command. This is optional
+                            and is set to a default value of DEFAULT_RESPONSE_TIMEOUT.
+
+        Returns:
+            True if the channel was homed.
+        """
+        _logger.info("[%s] Checking if channel %d is homed", self._name, channel_number)
+        self._validate_channel(channel_number)
+        self._check_is_open()
+        # This command needs a workaround. Instead of returning the MOT_MOVE_HOMED message, the instrument returns
+        # its current state first, so read that to discard the buffer
+        _ = self._apt_protocol.ask(MOT_GET_USTATUSUPDATE, timeout)
+        # then read the actual response we need
+        resp = self._apt_protocol.ask(MOT_MOVE_HOMED, timeout)
+
+        # Check if the channel number in the response is equal to the one that was asked for.
+        return resp.chan_ident == Thorlabs_MPC320_ChannelMap[channel_number]
```

Comment:
Question: I find this a little baffling (probably a Thorlabs bug): So the response is sent only when the move is completed and the returned "chan_ident" just says the channel number? This could receive an answer of another channel and fail, right?

Reply:
This was the case for `MOD_GET_CHANENABLESTATE` so I have do it for the others as well.
The response is only sent when the move is completed, which is why I now have a try except block. If it receives the info for another channel I would like it to fail, since I don't know whether the channel that I wanted info for is homed or not